.github/workflows/installers.yml

```diff
@@ -168,3 +174,27 @@ jobs:
         name: SasView-Installer-${{ matrix.os }}-${{ matrix.python-version }}-${{ github.sha }}
         path: installers/dist/SasView5.dmg
         if-no-files-found: error
+
+    - name: Declare env variables on push only
+      if: github.event_name == 'push'
+      shell: bash
+      run: echo "BRANCH_NAME=${GITHUB_REF#refs/*/}" >> $GITHUB_ENV
+
+    - name: Declare env variables on pull_request only
+      if: github.event_name == 'pull_request'
+      shell: bash
+      run: echo "BRANCH_NAME=$GITHUB_HEAD_REF" >> $GITHUB_ENV
+
+    - name: Upload installer to GitHub releases
+      #Release and main branch will be treated differently
+      if: env.BRANCH_NAME != 'main' && env.BRANCH_NAME != 'release*'
```

Comment:
The second part of the conditional is essentially not needed, since this workflow is running on non-release builds anyway.